2

```diff
@@ -13,6 +13,10 @@ done
 
 [ -x /etc/rc.local ] && /etc/rc.local
 
+for script in /etc/rc.local.d/??-*.sh; do
+    [ -x $script ] && $script
```

Comment:
What about passing a "start" or "stop" arg to allow one executable to do double duty?

Reply:
My 2 cents: I'd keep it simple and just execute without any arguments, leaving it to one-liner wrapper scripts to provide any args to downstream executables. 

Unless, of course, those specific "start" and "stop" args are a common interface that most existing executables already adhere to (and not other values like, say, "up" and "down",  in which case they'd still need a one-liner wrapper script).

Reply:
Ignore that last part,  it's called LSB interface.